File: cpp/646.maximum-length-of-pair-chain.cpp
class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {

        sort(pairs.begin(), pairs.end(), [&](const vector<int>& a, const vector<int>& b) {
            return a[1] < b[1];
        });

        int length = 1;
        int end = pairs[0][1];

        for (int i = 0; i < pairs.size(); i++) {
            if (end < pairs[i][0]) {
                end = pairs[i][1];
                length++;
            }
        }
        return length;
    }
};

File: python/646.maximum-length-of-pair-chain.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:

        pairs.sort(key=lambda x: x[1])
        n = len(pairs)
        length = 1  # it's guaranteed that we have one pair
        end = pairs[0][1]
        
        for i in range(1, n):
            if pairs[i][0] > end:
                length += 1
                end = pairs[i][1]

        return length